### PR TITLE
feat: set torch profiler in torch trials [DET-5549]

### PR DESCRIPTION
**Pull request by azhou-determined**
## Description

Add set profiler option to torch trial contexts allowing models to pass in profiler arguments.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: harness/determined/pytorch/_pytorch_trial.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def get_epoch_idx(self, batch_id: int) -> int:
         return batch_id // len(self.training_loader)
 
     def _average_training_metrics(
-        self, per_batch_metrics: List[Dict[str, Any]]
+            self, per_batch_metrics: List[Dict[str, Any]]
```
Comment (eecsliu) [inline harness/determined/pytorch/_pytorch_trial.py:217]: check linting - I wonder what's happening with these tabs.

**Review by eecsliu — COMMENTED** [review]
a bunch of wild tabs appeared

**Review by eecsliu — APPROVED** [review]
🚢
